src/System.IO.FileSystem.Watcher/src/System/IO/FileSystemWatcher.OSX.cs

```diff
@@ -191,224 +204,233 @@ private void CreateStreamAndStartWatcher()
                 _watcherThread.Start();
                 _runLoopStartedEvent.WaitOne();
```

Comment:
The thread should be made a background thread so that its existence doesn't prevent the process from shutting down.

And rather than making _runLoopStartedEvent a field on RunningInstance, you can just have it be a local here, and pass it into the Start method as an argument, then change the ThreadStart to be a ParameterizedThreadStart, and change WatchForFileSystemEventsThreadStart to take an argument, e.g.
```C#
var runLoopStarted = new ManualResetEventSlim();
new Thread(WatchForFileSystemEventsThreadStart) { IsBackground = true }.Start(runLoopStarted);
runLoopStarted.Wait();
...
private void WatchForFileSystemEventsThreadstart(object arg)
{
    var runLoopStarted = (ManualResetEventSlim)arg;
    ...
}
```
You could also use a Task instead of Thread if you wanted.